components/usage/pkg/apiv1/usage.go

```diff
@@ -32,6 +32,8 @@ type UsageService struct {
 
 	reportGenerator *ReportGenerator
 
+	lastReconciliation time.Time
```

Comment:
This assumes the service is a singleton and that the scheduled job will always run on the same instance. Is that a valid assumption?

Reply:
I think it's safe to assume that the usage service will remain a singleton (ie we won't run multiple replicas of it).

> and that the scheduled job will always run on the same instance

I'm not sure what you mean by this.

An alternative is to store this last reconciliation time in the database - then it will persist across pod restarts. I think the 🛹  is to do it in memory like this as the information returned is intended to be extra information to help make sense of the usage view.

Reply:
The controller doesn't run on the US clusters currently. How will this value be populated for US clusters?

Reply:
And related, what happens when the instance starts but the controller hasn't yet run?

Reply:
> The controller doesn't run on the US clusters currently. How will this value be populated for US clusters?

That's a good point. I hadn't considered that we no longer run the usage service in the US. That suggests that we need to write this to the database and rely on `db-sync` to keep the value consistent between both EU and US.

Reply:
> And related, what happens when the instance starts but the controller hasn't yet run?

In that case the method returns the timestamp corresponding to the zero value of a `time.Time` struct.